tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts

```diff
@@ -2456,6 +2456,64 @@ describe('scalar card', () => {
       ]);
     }));
 
+    it('builds range selected step data object', fakeAsync(() => {
+      const runToSeries = {
+        run1: [
+          {wallTime: 1, value: 1, step: 1},
+          {wallTime: 2, value: 10, step: 2},
+          {wallTime: 3, value: 20, step: 3},
+        ],
+        run2: [
+          {wallTime: 1, value: 1, step: 1},
+          {wallTime: 2, value: 10, step: 2},
+          {wallTime: 3, value: 25, step: 3},
+        ],
+      };
+      provideMockCardRunToSeriesData(
+        selectSpy,
+        PluginType.SCALARS,
+        'card1',
+        null /* metadataOverride */,
+        runToSeries
+      );
+      store.overrideSelector(
+        selectors.getCurrentRouteRunSelection,
+        new Map([
+          ['run1', true],
+          ['run2', true],
+        ])
+      );
+
+      store.overrideSelector(getMetricsLinkedTimeSelection, {
+        start: {step: 1},
+        end: {step: 3},
+      });
+
+      const fixture = createComponent('card1');
+      const scalarCardComponent = fixture.debugElement.query(
+        By.directive(ScalarCardComponent)
+      );
+      fixture.detectChanges();
+
+      const data =
+        scalarCardComponent.componentInstance.getTimeSelectionTableData();
+
+      expect(data).toEqual([
+        {
+          COLOR: '#fff',
+          RUN: 'run1',
+          STEP: 1,
+          VALUE_CHANGE: 19,
+        },
+        {
+          COLOR: '#fff',
+          RUN: 'run2',
+          STEP: 1,
+          VALUE_CHANGE: 24,
+        },
+      ]);
+    }));
+
     it('selects closest points to time selection', fakeAsync(() => {
```

Comment:
Should endStep be integrated into this test since there is also logic to handle finding the closets point? Or perhaps a new test for that? (I don't think this is covered in the new test you've already added).

Reply:
Yea, the problem is that currently we do not add that step into the data. So I could test it by ensuring that the VALUE_CHANGE is correct when the selected step is off. But, I think it will be cleaner to add this test when I add the END_STEP value. Which I will do next.